### PR TITLE
Closes #2886: Replace the existing "What Plan Do I Need" banner with the new one

### PR DESCRIPTION
**Pull request by jeawhanlee**
# Description

Fixes https://github.com/wp-media/imagify/issues/2886
Users now see all plans on the upgrade modal at first glance.

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).

## Detailed scenario

No Issue to reproduce

### Documentation

- Updated plans modal and the plan switcher

### New dependencies

N/A

### Risks

N/A

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.

## Code style
- [x] I wrote a self-explanatory code about what it does.
- [x] I did not introduce unnecessary complexity.

### CODE REVIEW

**Comment by jeawhanlee** [comment]
New Modal Design:
Screenshot 2024-10-03 at 17 27 39

Old Modal Design:
Screenshot 2024-10-03 at 17 28 59
Screenshot 2024-10-03 at 17 29 14

File: assets/css/pricing-modal.css [hunk 1]
```diff
@@ -1338,3 +1360,141 @@ p.imagify-modal-sidebar-title.imagify-modal-sidebar-title {
...
+.toggle-container img:nth-of-type(2) {
+    display: none;
+}
+input:checked ~ .badge-container .badge {
+    background-color: #97c45e;
+    color: white;
+    border-color: #97c45e;
+    opacity: 1; 
+}
+/** End Toggle Switch **/
```
Comment (remyperona) [inline assets/css/pricing-modal.css]: I would suggest to prefix all those classes with `imagify-` to make sure it doesn't conflict with anything else, the names are too generic currently

File: assets/js/pricing-modal.js [hunk 2]
```diff
@@ -31,6 +31,17 @@
...
+	$('#toggle-plan').change(function() {
+		var isChecked = $(this).is(':checked');
+		$('.toggle-label').eq(0).css('color', isChecked ? '#c8ced5' : '#3b3f4a');
+		$('.toggle-label').eq(1).css('color', isChecked ? '#3b3f4a' : '#c8ced5');
+		$('.badge').toggleClass('badge-checked', isChecked);
+		$('#imagify_all_plan_view').toggleClass('imagify-year-selected', isChecked).toggleClass('imagify-month-selected', ! isChecked);
+		$('.arrow-container img').eq(0).toggle(! isChecked);
+		$('.arrow-container img').eq(1).toggle(isChecked);
+	});
+
```
Comment (remyperona) [inline assets/js/pricing-modal.js]: prefix to be reflected here

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
NewBanner free:
newbanner_free

New popup:
newPopUp_2
newPopup_1

No green banner for growth and infinite user :heavy_check_mark:

**Comment by marija-nikolic** [comment]
Hi team, please make sure that we are using the arrow from Font Awesome 6 pro on the buttons. I've just double-checked what Célia (our designer) is using. 

Please check the pink upgrade banners too.

**Comment by jeawhanlee** [comment]
Both banners now having the expected arrow icon in button.

Screenshot 2024-10-08 at 14 01 47

Screenshot 2024-10-08 at 14 02 51

**Review by remyperona — APPROVED** [review]

**Comment by marija-nikolic** [comment]
> Both banners now having the expected arrow icon in button.
> 
> Screenshot 2024-10-08 at 14 01 47
> 
> Screenshot 2024-10-08 at 14 02 51

I am sorry @jeawhanlee, I missed this comment. The design of the pink banner is strange, it doesn't reflect what we should have. Could you please check? We validated everything with @Mai-Saad on the retreat.

**Comment by marija-nikolic** [comment]
> NewBanner free: newbanner_free
> 
> New popup: newPopUp_2 newPopup_1
> 
> No green banner for growth and infinite user ✔️

Sorry for the delay in checking the new popups. In the end we won't have any changes, the only thing we should improve is the alignment of /MONTH:
You can use the alignment on https://app.imagify.io/subscription/ as a reference:
image

**Comment by Mai-Saad** [comment]
GreenBanner
greenBanner
PinkBanner
pinkBanner
Adminbar when pinkBanner state (same already on production)
redadminbar_noImagifyPage

**Comment by Mai-Saad** [comment]
> Adminbar when pinkBanner state (same already on production)

This is fixed now on PR
AdminWIthremianing 20%

**Comment by Mai-Saad** [comment]
Merged dev to the PR local and here are screenshots of banners
**Green banner** 
Settings page
Screenshot from 2024-10-15 10-06-50

Bulk page
BulkPageRemaining 20%



**Pink banner**
settings page 
settingsRemaining20%

bulk page
Screenshot from 2024-10-15 10-40-37

**Click upgrade now popUp**
From admin bar link
Screenshot from 2024-10-15 10-43-06
Screenshot from 2024-10-15 10-45-38

From green banner it keeps loading (@jeawhanlee can you please check this :pray:)

upgrade_nowIn BulkPage

**Comment by Mai-Saad** [comment]
upgrade link from the green banner is working now
Screenshot from 2024-10-16 16-03-09
Screenshot from 2024-10-16 16-03-16

However, can see this in debug.log when click upgrade btn in admin bar  or green banner after fresh install `[16-Oct-2024 13:09:36 UTC] PHP Deprecated:  round(): Passing null to parameter #1 ($num) of type int|float is deprecated in /var/www/new.rocketlabsqa.ovh/htdocs/wp-content/plugins/imagify-plugin/inc/classes/class-imagify-files-stats.php on line 408` ===> added new GH about that as it's already on trunk https://github.com/wp-media/imagify-plugin/issues/909

**Pull request merged**